### ISSUE TITLE
std::move prevents copy ellision of std::function

### ISSUE THREAD

**Issue by codesaurus97** [bshoshany/thread-pool#3]
When using the lib, a compiler warning occurs due to moving and object directly and not a unique/shared pointer.
Place of occurrence: thread_pool.hpp:131

clang warning:
```
warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
            tasks.push(std::move(std::function<void()>(task)));

note: remove std::move call here
            tasks.push(std::move(std::function<void()>(task)));
```

Solution: Remove the std::move and push the object directly to the vector.

```c++
    template <typename F>
    void push_task(const F &task)
    {
        tasks_waiting++;
        {
            const std::scoped_lock lock(queue_mutex);
            tasks.push(std::function<void()>(task));
        }
    }
```

**Comment by bshoshany** [comment]
Thanks for the comment! Interestingly, GCC doesn't give me that warning even with `-Wall -Wextra -Wpedantic`, but Clang indeed does. Usually I debug with GCC, so I missed it.

The change you suggested does make sense, but just to be sure, I'll do some tests. If everything works as expected, it will be added to the next version.

**Comment by bshoshany** [comment]
Closing since this issue is now fixed in v1.3. Interestingly, the assembly code generated by the compiler seems to be the same before and after this change, presumably because the compiler eliminates the `std::move` automatically, but this change gets rid of the Clang warning.
